### PR TITLE
IGVF-445-reconcile-mouse-donor

### PR DESCRIPTION
**Pull request by dougli1sqrd**
Rodent Donor search and object page

### CODE REVIEW

File: components/product-info.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        {source.url ? (
+          <Link href={source.url}>{`${source.name} - ${lotProd}`}</Link>
+        ) : (
+          <>{`${source.name} - ${lotProd}`}</>
+        )}
+      </>
+    );
+  }
+
+  return <>{`${lotProd}`}</>;
```
Comment (forresttanaka) [inline components/product-info.js:22]: You can just use `<>{lotProd}</>`.

File: components/product-info.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from "prop-types";
+import Link from "next/link";
+
+// We want ideally: ABCAM - CD7348 - 11100020
+// <Link href="source.url">{source.name} - {productId} - {lotId}</Link>
+// If no url, then `{source.name} - {productId} - {lotId}`
+// We know if we have a lotId, then we have a productId
+export default function ProductInfo({ source, productId, lotId }) {
```
Comment (forresttanaka) [inline components/product-info.js:8]: None of these are required, so nice to have default values.

File: pages/rodent-donors/[uuid].js [hunk 3]
```diff
@@ -21,12 +21,16 @@ import buildBreadcrumbs from "../../lib/breadcrumbs";
...
+import PhenotypicFeatureTable from "../../components/phenotypic-feature-table";
+import ProductInfo from "../../components/product-info";
 
 export default function RodentDonor({
   donor,
   documents,
   parents,
   attribution = null,
+  phenotypicFeatures,
+  source,
```
Comment (forresttanaka) [inline pages/rodent-donors/[uuid].js:33]: Need default values for non-required properties.

**Review by forresttanaka — CHANGES_REQUESTED** [review]
Looks real good! I just have a couple really minor things for consistency, and you can take it directly to QA Ready after those are in.

**Comment by forresttanaka** [comment]
@dougli1sqrd Could you rebase this before moving to QA Ready? It has merge conflicts with dev.

File: components/search/list-renderer/rodent-donor.js [hunk 4]
```diff
@@ -22,9 +42,15 @@ export default function RodentDonor({ item: rodentDonor }) {
         <SearchListItemTitle>
           {rodentDonor.strain} {rodentDonor.sex}
         </SearchListItemTitle>
-        <SearchListItemMeta>
-          <div key="lab">{rodentDonor.lab.title}</div>
-        </SearchListItemMeta>
+        {(lab || collections) && (
```
Comment (forresttanaka) [inline components/search/list-renderer/rodent-donor.js:45]: Do you also need an `|| phenotypicFeatures` here, or do you not display the meta if you have no lab no collections even if you have phenotypic features?

**Review by forresttanaka — DISMISSED** [review]
Just one aspect I wasn’t certain about.

File: components/search/list-renderer/rodent-donor.js (hunk 4 above)
Comment (dougli1sqrd) [reply components/search/list-renderer/rodent-donor.js:45]: no that's true, I'll add it

**Review by forresttanaka — DISMISSED** [review]
😎

File: components/product-info.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ export default function ProductInfo({
 }) {
   const prodLot = lotId ? `${productId} (${lotId})` : "";
   if (source) {
+    console.log(source);
```
Comment (forresttanaka) [inline components/product-info.js]: Need to delete this.

**Review by forresttanaka — APPROVED** [review]
Looks like that should get rid of the “undefined undefined” issue in the phenotype table.

**Pull request merged**